anvio/parsers/hmmscan.py

```diff
@@ -49,20 +66,76 @@ def __init__(self, hmm_scan_hits_txt, alphabet='AA', context='GENE'):
         Parser.__init__(self, 'HMMScan', [hmm_scan_hits_txt], files_expected, files_structure)
 
 
-    def get_search_results(self):
+    def get_search_results(self, ko_list_dict = None):
```

Comment:
I don't think we should call this variable `ko_list_dict`. this basically is a dictionary that describes noise cutoff thresholds for each HMM individually. it can be used for KEGG KOs, but the concept is more general than KEGG (and this module doesn't know about KEGG, so we don't have to be too specific).

So the variable `ko_list_dict` can be renamed to `noise_cutoff_dict`, `knum` to `hmm_entry_name`, etc. Furthermore, error messages should be agnostic to KEGG / KO, too.